hoomd/md/alchemy/pair.py

```diff
@@ -0,0 +1,368 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Alchemical pair forces."""
+from collections.abc import Mapping
+
+import hoomd.data
+from hoomd.logging import log
+from hoomd.operation import _HOOMDBaseObject
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.md.pair import LJGauss as BaseLJGauss
+
+
+def _modify_pair_cls_to_alchemical(cls):
+    """Modifies a created class inheriting from `_AlchemicalPairForce`.
+
+    This decorator sets the _dof_cls type, updates the ``_cpp_class_name``,
+    ``_accepted_modes``, and ``_reserved_default_attrs``, and sets ``normalize =
+    False`` if not set.
+    """
+    new_cpp_name = [
+        'PotentialPair', 'Alchemical', cls.__mro__[0]._cpp_class_name[13:]
+    ]
+    if getattr(cls, 'normalized', False):
+        new_cpp_name.insert(2, 'Normalized')
+        cls._dof_cls = AlchemicalNormalizedDOF
+    else:
+        cls.normalized = False
+        cls._dof_cls = AlchemicalDOF
+    cls._cpp_class_name = ''.join(new_cpp_name)
+    cls._accepted_modes = ('none', 'shift')
+    return cls
+
+
+class AlchemicalPairDOFStore(Mapping):
```

Comment:
This class is not directly used by users. I will make it private unless there are objections.

Reply:
No objections here.

Reply:
It is used by users, every time `ljg.r0` is accessed the user uses it. We can make it private but we have made `TypeParameter` public which is its equivalent.